cfg/LocalRef.cc

```diff
@@ -0,0 +1,70 @@
+#include "cfg/LocalRef.h"
+#include "cfg/CFG.h"
+
+using namespace std;
+
+namespace sorbet::cfg {
+core::LocalVariable LocalRef::data(const CFG &cfg) const {
+    return cfg.localVariables[this->_id];
+}
+
+int LocalRef::minLoops(const CFG &cfg) const {
+    ENFORCE(cfg.minLoops.size() > this->_id);
+    ENFORCE(this->exists());
+    return cfg.minLoops[this->_id];
+}
+
+int LocalRef::maxLoopWrite(const CFG &cfg) const {
+    ENFORCE(cfg.maxLoopWrite.size() > this->_id);
+    ENFORCE(this->exists());
+    return cfg.maxLoopWrite[this->_id];
+}
+
+bool LocalRef::exists() const {
+    return this->_id > 0;
+}
+
+bool LocalRef::isAliasForGlobal(const core::GlobalState &gs, const CFG &cfg) const {
+    return this->data(cfg).isAliasForGlobal(gs);
+}
+
+bool LocalRef::isSyntheticTemporary(const core::GlobalState &gs, const CFG &cfg) const {
+    return this->data(cfg).isSyntheticTemporary(gs);
+}
+
+string LocalRef::toString(const core::GlobalState &gs, const CFG &cfg) const {
+    return this->data(cfg).toString(gs);
+}
+
+string LocalRef::showRaw(const core::GlobalState &gs, const CFG &cfg) const {
+    return this->data(cfg).showRaw(gs);
+}
+
+bool LocalRef::operator==(const LocalRef &rhs) const {
+    return this->_id == rhs._id;
+}
+
+bool LocalRef::operator!=(const LocalRef &rhs) const {
+    return this->_id != rhs._id;
+}
+
+LocalRef LocalRef::noVariable() {
+    return LocalRef(0);
```

Comment:
These magic values are ENFORCE-d in sync in the constructor for CFG, which is similar to what we do for SymbolRefs.

Reply:
Could you put a note to this effect in LocalRef.h, and also in the `.data()` method here? I'd like to not lose track of the fact that these are expected to exist in the CFG.